tools/dart_tools/bin/fetch_glfw.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:archive/archive.dart';
+
+const String glfwArchiveUrlString =
+    'http://github.com/glfw/glfw/releases/download/3.2.1/glfw-3.2.1.bin.WIN64.zip';
+
+Future<void> main(List<String> arguments) async {
+  if (!Platform.isWindows) {
+    throw new Exception('Prebuilt glfw3 libraries are only available on '
+        'windows.');
+  }
+
+  if (arguments.length != 1) {
+    throw new Exception('Only one argument should be passed, the directory to '
+        'download glfw to.');
+  }
+
+  final outputDirectory = arguments[0];
+
+  const Map<String, String> requiredFiles = {
+    'glfw-3.2.1.bin.WIN64/include/GLFW/glfw3.h': 'glfw3.h',
+    'glfw-3.2.1.bin.WIN64/lib-vc2015/glfw3.lib': 'glfw3.lib',
+  };
+
```

Comment:
Even though this file is relatively short, it would benefit from functions that abstract away the details of each distinct step. The individual steps that I would recommend pulling out are:
- Checking whether the download needs to happen (returning a bool)
- Downloading the zip data
- Extracting the zip data (a helper called by the download function)

That would parallel the structure of update_engine_hash, making it easier for someone who has read one to easily read/understand the other.